rinja_derive/src/lib.rs

```diff
@@ -231,6 +232,30 @@ impl fmt::Display for FileInfo<'_> {
     }
 }
 
+struct MsgValidEscapers<'a>(&'a [(Vec<Cow<'a, str>>, Cow<'a, str>)]);
+
+impl fmt::Display for MsgValidEscapers<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let exts: BTreeSet<&str> = self
+            .0
+            .iter()
+            .flat_map(|(exts, _)| exts)
+            .map(|x| x.as_ref())
+            .collect();
+        f.write_str("The available extensions are:")?;
+        for (i, mut e) in exts.iter().copied().enumerate() {
+            if e.is_empty() {
+                e = r#""" <empty string>"#;
```

Comment:
I thought it was a debug display before checking. XD

I think using parens would be better here:

```suggestion
                e = r#""" (empty string)"#;
```

One extra thought: wouldn't it be better to rely on `join` instead? So your first iterator could become:

```rust
let exts: String = self
            .0
            .iter()
            .flat_map(|(exts, _)| exts)
            .map(|x| if x.is_empty() { r#""" (empty string)"# else { x.as_ref() })
            .collect::<BTreeSet<&str>>()
            .into_iter()
            .join(", ");
```

No need for the loop below like this. And I think perf is not really a big issue here since we're errorring.

Reply:
You're right. :) I use `slice::join` and the debug display of the extension, now.